joint_state_broadcaster/src/joint_state_broadcaster.cpp

```diff
@@ -351,7 +351,9 @@ controller_interface::return_type JointStateBroadcaster::update(
       state_interface.get_value());
   }
 
-  if (realtime_joint_state_publisher_ && realtime_joint_state_publisher_->trylock())
+  if (
+    realtime_joint_state_publisher_ && joint_state_publisher_->get_subscription_count() > 0u &&
```

Comment:
looks good. Are the `rmw` and implementations thereof doing this implicitly already?

Reply:
I believe yes

Reply:
meaning subscription count is polled outside the RT loop where it is definitely safe, right? To my best understanding, the `publish` operation just appends a buffer